E2E dual cluster test failed intermittently

### Describe the bug



https://github.com/hiero-ledger/solo/actions/runs/26967134460/job/79573637617?pr=4511

```

  1) Dual Cluster Full E2E Test Suite
       Dual Cluster Full E2E Test
         dual-cluster-full: mirror node add:
     Uncaught AssertionError: expect the created account to exist in the mirror nodes copy of the accounts: expected undefined to equal '3.2.1001'
      at IncomingMessage.<anonymous> (/home/runner/_work/solo/solo/test/e2e/commands/tests/mirror-node-test.ts:213:22)
      at IncomingMessage.emit (node:events:508:28)
      at Readable.read (node:internal/streams/readable:784:10)
      at flow (node:internal/streams/readable:1290:53)
      at resume_ (node:internal/streams/readable:1269:3)
      at process.processTicksAndRejections (node:internal/process/task_queues:90:21)

  2) Dual Cluster Full E2E Test Suite
       Dual Cluster Full E2E Test
         "after all" hook for "dual-cluster-full: consensus network destroy should success":
     Uncaught AssertionError: expect the created account to exist in the mirror nodes copy of the accounts: expected undefined to equal '3.2.1001'
      at IncomingMessage.<anonymous> (/home/runner/_work/solo/solo/test/e2e/commands/tests/mirror-node-test.ts:213:22)
      at IncomingMessage.emit (node:events:508:28)
      at Readable.read (node:internal/streams/readable:784:10)
      at flow (node:internal/streams/readable:1290:53)
      at resume_ (node:internal/streams/readable:1269:3)
      at process.processTicksAndRejections (node:internal/process/task_queues:90:21)

```

### Describe the expected behavior

Should not fail

### To Reproduce

PR checks

### Additional Context

_No response_

https://github.com/hiero-ledger/solo/actions/runs/27018317153/job/79740163432?pr=4511

```
  1) Dual Cluster Full E2E Test Suite
       Dual Cluster Full E2E Test
         dual-cluster-full: mirror node add:
     MirrorNodeDeployFailedSoloError: Error deploying mirror node: transaction 3.2.2@1780667871.459204200 failed precheck with status INVALID_NODE_ACCOUNT against node account id 3.2.3
      at MirrorNodeCommand.add (/home/runner/_work/solo/solo/src/commands/mirror-node.ts:1421:15)
      at process.processTicksAndRejections (node:internal/process/task_queues:104:5)
      at async Object.handler (/home/runner/_work/solo/solo/src/core/command-path-builders/command-builder.ts:129:47)
     Caused by: StatusError: transaction 3.2.2@1780667871.459204200 failed precheck with status INVALID_NODE_ACCOUNT against node account id 3.2.3
      at new PrecheckStatusError (/home/runner/_work/solo/solo/node_modules/@hiero-ledger/sdk/lib/PrecheckStatusError.cjs:48:5)
      at FileContentsQuery._mapStatusError (/home/runner/_work/solo/solo/node_modules/@hiero-ledger/sdk/lib/query/Query.cjs:423:12)
      at FileContentsQuery.execute (/home/runner/_work/solo/solo/node_modules/@hiero-ledger/sdk/lib/Executable.cjs:795:22)
      at process.processTicksAndRejections (node:internal/process/task_queues:104:5)
      at async AccountManager.prepareAddressBookBase64 (/home/runner/_work/solo/solo/src/core/account-manager.ts:1087:24)
      at async Task.task (/home/runner/_work/solo/solo/src/commands/mirror-node.ts:974:42)
      at async Task.run (/home/runner/_work/solo/solo/node_modules/listr2/dist/index.mjs:1908:5)

```

Another incident

https://github.com/hiero-ledger/solo/actions/runs/27282954684/job/80583346302

```
  1 failing

  1) Dual Cluster Full E2E Test Suite
       Dual Cluster Full E2E Test
         dual-cluster-full: mirror node add:
     MirrorNodeDeployFailedSoloError: Error deploying mirror node: transaction 3.2.2@1781102247.241610231 failed precheck with status INVALID_NODE_ACCOUNT against node account id 3.2.3
      at MirrorNodeCommand.add (/home/runner/_work/solo/solo/src/commands/mirror-node.ts:1421:15)
      at process.processTicksAndRejections (node:internal/process/task_queues:104:5)
      at async Object.handler (/home/runner/_work/solo/solo/src/core/command-path-builders/command-builder.ts:129:47)
     Caused by: StatusError: transaction 3.2.2@1781102247.241610231 failed precheck with status INVALID_NODE_ACCOUNT against node account id 3.2.3
      at new PrecheckStatusError (/home/runner/_work/solo/solo/node_modules/@hiero-ledger/sdk/lib/PrecheckStatusError.cjs:48:5)
      at FileContentsQuery._mapStatusError (/home/runner/_work/solo/solo/node_modules/@hiero-ledger/sdk/lib/query/Query.cjs:423:12)
      at FileContentsQuery.execute (/home/runner/_work/solo/solo/node_modules/@hiero-ledger/sdk/lib/Executable.cjs:795:22)
      at process.processTicksAndRejections (node:internal/process/task_queues:104:5)
      at async AccountManager.prepareAddressBookBase64 (/home/runner/_work/solo/solo/src/core/account-manager.ts:1080:24)
      at async Task.task (/home/runner/_work/solo/solo/src/commands/mirror-node.ts:974:42)
      at async Task.run (/home/runner/_work/solo/solo/node_modules/listr2/dist/index.mjs:1908:5)


```

--- a/src/core/account-manager.ts
+++ b/src/core/account-manager.ts
@@ -21,6 +21,7 @@ import {
   Logger,
   LogLevel,
   Long,
+  PrecheckStatusError,
   PrivateKey,
   Status,
   TransactionReceipt,
@@ -1074,10 +1075,36 @@ export class AccountManager {
 
     const realm: Realm = this.localConfig.configuration.realmForDeployment(deployment);
     const shard: Shard = this.localConfig.configuration.shardForDeployment(deployment);
-    const query: FileContentsQuery = new FileContentsQuery().setFileId(
-      new FileId(shard, realm, FileId.ADDRESS_BOOK.num),
-    );
-    return Buffer.from(await query.execute(client)).toString('base64');
+    const fileId: FileId = new FileId(shard, realm, FileId.ADDRESS_BOOK.num);
+
+    // The SDK does not retry INVALID_NODE_ACCOUNT for queries (only for transactions), so we
+    // retry here to handle the race where a node's gRPC endpoint is up but its Hedera state
+    // has not finished initializing yet.
+    const maxAttempts: number = 5;
+    const retryDelayMs: number = 5000;
+    let lastError: PrecheckStatusError | undefined;
+
+    for (let attempt: number = 0; attempt < maxAttempts; attempt++) {
+      try {
+        return Buffer.from(await new FileContentsQuery().setFileId(fileId).execute(client)).toString('base64');
+      } catch (error) {
+        if (
+          error instanceof PrecheckStatusError &&
+          error.status === Status.InvalidNodeAccount &&
+          attempt < maxAttempts - 1
+        ) {
+          this.logger.warn(
+            `Address book query returned INVALID_NODE_ACCOUNT (attempt ${attempt + 1}/${maxAttempts}), retrying in ${retryDelayMs}ms`,
+          );
+          lastError = error;
+          await sleep(Duration.ofMillis(retryDelayMs));
+        } else {
+          throw error;
+        }
+      }
+    }
+
+    throw lastError!;
   }
 
   public async getFileContents(

